packages/blockchain-api/src/currencyConversion/CurrencyConversionAPI.test.ts

```diff
@@ -0,0 +1,134 @@
+
+describe('CurrencyConversionAPI', () => {
+  let currencyConversionAPI: CurrencyConversionAPI
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    currencyConversionAPI = new CurrencyConversionAPI()
+    currencyConversionAPI.initialize({ context: {}, cache: new InMemoryLRUCache() })
+  })
+
+  it('should retrieve rate for cGLD/cUSD', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'cGLD',
+      currencyCode: 'cUSD',
+    })
+    expect(result).toEqual(new BigNumber(10))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(0)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retrieve rate for cUSD/cGLD', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'cUSD',
+      currencyCode: 'cGLD',
+    })
+    expect(result).toEqual(new BigNumber(10))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(0)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retrieve rate for cGLD/USD', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'cGLD',
+      currencyCode: 'USD',
+    })
+    expect(result).toEqual(new BigNumber(10))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(0)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retrieve rate for USD/cGLD', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'USD',
+      currencyCode: 'cGLD',
+    })
+    expect(result).toEqual(new BigNumber(10))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(0)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retrieve rate for cGLD/MXN', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'cGLD',
+      currencyCode: 'MXN',
+    })
+    expect(result).toEqual(new BigNumber(200))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(1)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retrieve rate for MXN/cGLD', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'MXN',
+      currencyCode: 'cGLD',
+    })
+    expect(result).toEqual(new BigNumber(200))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(1)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retrieve rate for USD/MXN', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'USD',
+      currencyCode: 'MXN',
+    })
+    expect(result).toEqual(new BigNumber(20))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(1)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(0)
+  })
+
+  it('should retrieve rate for MXN/USD', async () => {
```

Comment:
Same with this one- is there anything this test would catch that wouldn't be covered by the `USD/MXN` one? 
Could also consider adding some tests for `getConversionSteps`

Reply:
I decided not to test `getConversionSteps` directly to avoid testing the internals too much.